src/subshell/common.c

```diff
@@ -1351,20 +1416,50 @@ subshell_name_quote (const char *s)
     // Copy the beginning of the command to the buffer
     g_string_append (ret, quote_cmd_start);
 
-    /*
-     * Print every character except digits and letters as a backslash-escape
-     * sequence of the form \0nnn, where "nnn" is the numeric value of the
-     * character converted to octal number.
-     */
+    /* Sending physical lines over a certain small limit causes problems on some platforms,
+     * see ticket #4480. Make sure to wrap in time. See how large we can grow so that an
+     * additional line wrapping or closing string still fits. */
+    const int max_length =
+        COOKED_MODE_BUFFER_SIZE - MAX (strlen (before_wrap), strlen (quote_cmd_end));
```

Comment:
it would be kinda more elegant to wrap the trailer separately if the situation occurs, but the extra code is probably not worth it ...

Reply:
if you assign the lengths and actually use them to pre-calculate and check the buffer, it's easier to make sure that you're actually working with the same numbers. the mismatch risk is real, but i personally worry about it less, because i have annoyingly good visual pattern recognition.

---

the length of the final cd command is currently subtracted from the buffer size, so it always fits. but it would be possible to ensure that only the line termination fits, and put the cd on its own line if necessary.